extensions/api/data-management/contractdefinition/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/contractdefinition/service/ContractDefinitionServiceImpl.java

```diff
@@ -0,0 +1,101 @@
+/*
+ *  Copyright (c) 2022 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ */
+
+package org.eclipse.dataspaceconnector.api.datamanagement.contractdefinition.service;
+
+import org.eclipse.dataspaceconnector.api.result.ServiceResult;
+import org.eclipse.dataspaceconnector.dataloading.ContractDefinitionLoader;
+import org.eclipse.dataspaceconnector.spi.contract.negotiation.store.ContractNegotiationStore;
+import org.eclipse.dataspaceconnector.spi.contract.offer.store.ContractDefinitionStore;
+import org.eclipse.dataspaceconnector.spi.query.Criterion;
+import org.eclipse.dataspaceconnector.spi.query.QuerySpec;
+import org.eclipse.dataspaceconnector.spi.transaction.TransactionContext;
+import org.eclipse.dataspaceconnector.spi.types.domain.contract.offer.ContractDefinition;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static java.lang.String.format;
+import static java.util.stream.Collectors.toList;
+
+public class ContractDefinitionServiceImpl implements ContractDefinitionService {
+    private final ContractDefinitionStore store;
+    private final ContractDefinitionLoader loader;
+    private final TransactionContext transactionContext;
+    private final ContractNegotiationStore contractNegotiationStore;
+
+    public ContractDefinitionServiceImpl(ContractDefinitionStore store, ContractDefinitionLoader loader, TransactionContext transactionContext, ContractNegotiationStore contractNegotiationStore) {
+        this.store = store;
+        this.loader = loader;
+        this.transactionContext = transactionContext;
+        this.contractNegotiationStore = contractNegotiationStore;
+    }
+
+    @Override
+    public ContractDefinition findbyId(String contractDefinitionId) {
+        var querySpec = QuerySpec.Builder.newInstance()
```

Comment:
We might need to add a `findById` method to the `ContractDefinitionStore`, because SQL-based stores may not have the query feature fully implemented for M3, just paging. No action to be taken on your part, just an FYI.

Reply:
Yes, will open an issue for this

Reply:
#967